ext/drwrap/drwrap.c

```diff
@@ -593,6 +597,8 @@ drwrap_get_mcontext_internal(drwrap_context_t *wrapcxt, dr_mcontext_flags_t flag
                  */
 #ifdef AARCHXX
                 wrapcxt->mc->xflags = 0; /*0 is fine for ARM */
+#elif defined(RISCV64)
+                wrapcxt->mc->xflags = 0; /* FIXME i#3544: Check if valid */
```

Comment:
If there are no condition codes (or other types of status codes for integer code) there shouldn't even be a field called `xflags`?

Reply:
I wasn't sure how to solve this. RISC-V indeed does not have condition codes. Instead all conditional branches are arithmetic register comparison based and those are also used for overflow checking.
`xflags` is supposed to be "platform independent condition flags" so question is should it be available on all architectures, just never updated on some? Removing it would add more ifdefs to client code, then again keeping it would give false impression of this field being functional.

Second point is that  there is a status and control register for floating point operations (`fcsr`) on RISC-V. Aarch64 keeps its own floating point registers in the machine context so I believe I should do the same. However currently I am aliasing xflags with it in mcxtx_api.h. Perhaps instead I should add a `dummy` field to mark non-existent condition flags instead?

Reply:
I would vote for removing the xflags field even if it means adding some ifdefs (in a separate PR).  From separate discussion @abhinav92003 agrees.

Re: floating-point status + control: that should be a separate field.  (AArch64 has FPCR and FPSR in there; x86 might be lacking but should be fixed: #3187).

Reply:
Done. xflags removal is in #5587.